pkgs/by-name/ml/mlx42/package.nix

```
@@ -15,32 +16,43 @@ stdenv.mkDerivation (finalAttrs: {
   src = fetchFromGitHub {
     owner = "codam-coding-college";
     repo = "MLX42";
-    rev = "v${finalAttrs.version}";
+    rev = "refs/tags/v${finalAttrs.version}";
     hash = "sha256-c4LoTePHhQeZTx33V1K3ZyXmT7vjB6NdkGVAiSuJKfI=";
   };
 
-  postPatch = ''
-    patchShebangs ./tools
-  ''
-  + lib.optionalString enableShared ''
-    substituteInPlace CMakeLists.txt \
-        --replace "mlx42 STATIC" "mlx42 SHARED"
-  '';
+  postPatch =
+    ''
+      patchShebangs --build ./tools
+    ''
+    + lib.optionalString enableShared ''
+      substituteInPlace CMakeLists.txt \
+          --replace-fail "mlx42 STATIC" "mlx42 SHARED"
+    '';
+
+  strictDeps = true;
 
   nativeBuildInputs = [ cmake ];
 
-  buildInputs = [ glfw ]
-    ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [ OpenGL Cocoa IOKit ]);
+  buildInputs =
+    [ glfw ]
+    ++ lib.optionals stdenv.isDarwin (
+      with darwin.apple_sdk.frameworks;
+      [
+        OpenGL
+        Cocoa
+        IOKit
+      ]
+    );
 
-  cmakeFlags = [ "-DDEBUG=${toString enableDebug}" ];
+  cmakeFlags = [ (lib.cmakeBool "DEBUG" enableDebug) ];
 
   postInstall = ''
     mkdir -p $out/lib/pkgconfig
     substituteAll ${./mlx42.pc} $out/lib/pkgconfig/mlx42.pc
   '';
 
   meta = {
-    changelog = "https://github.com/codam-coding-college/MLX42/releases/tag/${finalAttrs.src.rev}";
+    changelog = "https://github.com/codam-coding-college/MLX42/releases/tag/${finalAttrs.version}";
```

```suggestion
    changelog = "https://github.com/codam-coding-college/MLX42/releases/tag/v${finalAttrs.version}";
```

version doesn't have `v` prefix

---

Ah, you're right. Fixed.